course/src/isometry.cc

```diff
@@ -308,5 +334,96 @@ const Matrix3 Matrix3::kZero =
 const Matrix3 Matrix3::kOnes = 
     Matrix3(Vector3(1, 1, 1), Vector3(1, 1, 1), Vector3(1, 1, 1));
 
+// Isometry
+Isometry::Isometry(const Matrix3& rotation) {
+    rotation_ = rotation;
+    translation_ = Vector3(0.0, 0.0, 0.0);
+}
+
+Isometry::Isometry(const Vector3& translation, const Matrix3& rotation) {
+    rotation_ = rotation;
+    translation_ = translation;
+}
+
+Isometry Isometry::compose(const Isometry& isometry) const {
+    Isometry aux(*this);
+    return Isometry(*this) = aux * isometry;
+};
+
+Isometry Isometry::inverse() const {
+    auto det = rotation_.det();
+    if (almost_equal(det, 0., 4)) {
+        throw std::domain_error("It can not get the inverse of the matrix");
+    }
+    Matrix3 rot_result;
+    rot_result.r1().x() = (rotation_.r2().y() * rotation_.r3().z()) - (rotation_.r3().y() * rotation_.r2().z());
+    rot_result.r1().y() = (rotation_.r1().z() * rotation_.r3().y()) - (rotation_.r3().z() * rotation_.r1().y());
+    rot_result.r1().z() = (rotation_.r1().y() * rotation_.r2().z()) - (rotation_.r2().y() * rotation_.r1().z());
+
+    rot_result.r2().x() = (rotation_.r2().z() * rotation_.r3().x()) - (rotation_.r3().z() * rotation_.r2().x());
+    rot_result.r2().y() = (rotation_.r1().x() * rotation_.r3().z()) - (rotation_.r3().x() * rotation_.r1().z());
+    rot_result.r2().z() = (rotation_.r1().z() * rotation_.r2().x()) - (rotation_.r2().z() * rotation_.r1().x());
+
+    rot_result.r3().x() = (rotation_.r2().x() * rotation_.r3().y()) - (rotation_.r3().x() * rotation_.r2().y());
+    rot_result.r3().y() = (rotation_.r1().y() * rotation_.r3().x()) - (rotation_.r3().y() * rotation_.r1().x());
+    rot_result.r3().z() = (rotation_.r1().x() * rotation_.r2().y()) - (rotation_.r2().x() * rotation_.r1().y());
+
+    rot_result = rot_result / det;
+    auto vector_result = rot_result * translation_;
+    vector_result = vector_result * (-1);
+
+    return Isometry(vector_result, rot_result);
+}
+
+Isometry Isometry::FromTranslation(const std::initializer_list<double>& values) {
+    return Isometry(Vector3(values), Matrix3::kIdentity);
+}
+
+Isometry Isometry::FromEulerAngles(double yaw, double pitch, double roll) {
+    Isometry result = Isometry(RotateAround(Vector3::kUnitX, yaw)) * Isometry(RotateAround(Vector3::kUnitY, pitch)) *
```

Comment:
You can get the rotation matrix from yaw, pitch and roll and create the isometry only once.